test/test-utils/wrappers.tsx

```diff
@@ -14,16 +14,16 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React, { RefCallback } from "react";
+import React, { RefCallback, ComponentType } from "react";
 import { MatrixClient } from "matrix-js-sdk/src/matrix";
 
 import { MatrixClientPeg as peg } from '../../src/MatrixClientPeg';
 import MatrixClientContext from "../../src/contexts/MatrixClientContext";
 
-type WrapperType<T> = React.Component<{ wrappedRef?: RefCallback<T> }>;
+type WrapperProps<T> = { wrappedRef?: RefCallback<ComponentType<T>> } & T;
 
-export function wrapInMatrixClientContext<T>(WrappedComponent): WrapperType<T> {
-    class Wrapper extends React.Component<{ wrappedRef?: RefCallback<T> }> {
+export function wrapInMatrixClientContext<T>(WrappedComponent: ComponentType<T>): ComponentType<WrapperProps<T>> {
+    class Wrapper extends React.Component<WrapperProps<T>> {
```

Comment:
Wonder why we're not just using a Functional component here

Reply:
I tried it, and there are apparently some tests that depend on this being a class component - namely EventListSummary-test which uses ReactTestUtils rather than Enzyme, and so depends on getting an instance to interact with, and TextualBody-test, which uses a `.setState({})` hack on the wrapper to work around waiting for state updates. I'd rather not worry about those here.